server/command.go

```diff
@@ -174,6 +174,84 @@ func (p *Plugin) executeCommandQueue(args *model.CommandArgs) *model.CommandResp
 	return &model.CommandResponse{}
 }
 
+func calculateQueItemNumber(args *model.CommandArgs, p *Plugin, hashtag string) (*model.CommandResponse, int) {
+	searchResults, appErr := p.API.SearchPostsInTeamForUser(args.TeamId, args.UserId, model.SearchParameter{Terms: &hashtag})
+	if appErr != nil {
+		return responsef("Error calculating list number"), 0
+	}
+	postList := *searchResults.PostList
+	numQueueItems := len(postList.Posts)
+	return nil, numQueueItems + 1
+}
+
+func (p *Plugin) executeCommandReQueue(args *model.CommandArgs) *model.CommandResponse {
+	split := strings.Fields(args.Command)
+
+	if len(split) <= 2 {
+		return responsef("Missing parameters for requeue command")
+	}
+
+	meeting, err := p.GetMeeting(args.ChannelId)
+	if err != nil {
+		return responsef("Can't find the meeting")
+	}
+
+	oldPostID := split[2]
+	postToBeReQueued, _ := p.API.GetPost(oldPostID)
+	var (
+		prefix            string
+		hashtagDateFormat string
+	)
+	if matchGroups := meetingDateFormatRegex.FindStringSubmatch(meeting.HashtagFormat); len(matchGroups) == 4 {
+		prefix = matchGroups[1]
+		hashtagDateFormat = strings.TrimSpace(matchGroups[2])
+	}
+
+	var (
+		messageRegexFormat = regexp.MustCompile(fmt.Sprintf(`(?m)^#### #%s(?P<date>.*) [0-9]+\) (?P<message>.*)?$`, prefix))
```

Comment:
Can we define this regex at the top of this file?

Reply:
0/5, not optimal since we substitute another value in regex here.

Reply:
We should use `regexp.Compile` instead of `MustCompile` so we can handle the error and avoid a panic. As well as anywhere else we are calling `MustCompile` inside of a function.